### DISCUSSION TITLE
Re-licencing Pulsar

### DISCUSSION THREAD

**Discussion by Daeraxa** [pulsar-edit/.github#72] [General]
I want to preface this by saying I'm **not** advocating for this one way or the other but it is almost an inevitable conversation so thought it would be worth discussing it early.

Currently Pulsar is licenced as MIT as inherited from Atom which is obviously one of the most permissive licences out there and really doesn't have any restrictions. This makes perfect sense for node packages, libraries and when big companies want to open source a project as it will allow them to do whatever they like down the line if they decide to close it.

However for projects like this does it make sense? It isn't like Pulsar itself would be integrated into another product and there is no large company behind it so I can see the justification in using something more restrictive like GPL or MPL where any forks would need to disclose their source code so that the original project could still benefit.

Now Pulsar does have its complexities, for example would it just be the core editor and core packages that get re-licenced but leave the other repo packages as MIT?

If nothing else I think it is useful to have the conversation so we can at least point people to it if it comes up in the future.

By way of comparison, a list of other editors sorted by permissiveness:
- Kakoune - The Unlicense
- Brackets - MIT
- LiteXL - MIT
- VSCode - MIT (except the binaries)
- Lapce - Apache 2.0
- Helix - MPL
- CudaText - MPL
- Kate - LGPL/GPL
- Vim - GPL compatible
- Neovim - Apache 2.0 (+ Vim's GPL)
- Geany - GPL 2.0
- Emacs - GPL 3.0 (duh...)
- TextMate - GPL 3.0
- Notepad++ - GPL 3.0

**Comment by confused-Techie** [comment]
Personally I prefer using MIT licensing. I know it means someone in the future could essentially close source further progression, but I like the open-ness. It's basically here is our code, and do whatever you want, which I feel is in line with a lot of our other values. But is a good idea to bring up

**Comment by sertonix** [comment]
It may not be on topic but it is related.
On some repository the license is set to MIT in the `package.json` but there is no license file. Should we add it there?
Also any preference of `LICENSE` vs. `LICENSE.md`?

**Reply by Daeraxa** [reply]
There was a discussion on this before as I still have a PR that I reverted and haven't gone back to where I re-licenced all the packages internal to the pulsar repo. Basically there is no need to licence anything that doesn't have one as it will be covered by the project level licence.

This is the commit in question: https://github.com/pulsar-edit/pulsar/pull/171/commits/e03eeb03e2b480f947c1f57030a9d64a93f76a62
Just forgot about it and was going to make a new PR with this included

**Reply by sertonix** [reply]
Ok, now I see potential for less files. If license is already defined top level (with the same content) can we remove it?

**Reply by Daeraxa** [reply]
Any that are identical, yes, I was planning to re-do this to remove any of the GH/Atom ones but there are a bunch with additional attributions which need to be maintained.

**Reply by sertonix** [reply]
> There was a discussion on this before as I still have a PR that I reverted and haven't gone back to where I re-licenced all the packages internal to the pulsar repo. Basically there is no need to licence anything that doesn't have one as it will be covered by the project level licence.

Now I a not sure anymore if you mean the same I mean. I mean packages that are in there own repository. eg. [command-palette](https://github.com/pulsar-edit/command-palette)

**Reply by Spiker985** [reply]
> > There was a discussion on this before as I still have a PR that I reverted and haven't gone back to where I re-licenced all the packages internal to the pulsar repo. Basically there is no need to licence anything that doesn't have one as it will be covered by the project level licence.
> 
> Now I a not sure anymore if you mean the same I mean. I mean packages that are in there own repository. eg. [command-palette](https://github.com/pulsar-edit/command-palette)

Basically, any repo which doesn't have a license.md, should ~~receive~~ inherit the .github license.md

Any repo which has a license.md, which matches the .github license, can be removed

Any repo that has a license.md, which does NOT match the .github license, should remain as it was - or at least be evaluated

**Reply by sertonix** [reply]
Ok, but where is the org license? I don't see it in [.github](https://github.com/pulsar-edit/.github).
